### DISCUSSION TITLE
Consider casting the result of `getpoint/3` to the band format of the subject image?

### DISCUSSION THREAD

**Discussion by kipcole9** [akash-akya/vix#153] [General]
Today, `Vix.Vips.Operation.getpoint/3` always returns a float values. This means that casting the result to the band format of the subject image is up to the caller. 

Typically this requires a call to `Vix.Vips.Image.format/1` and therefore an additional NIF call.

I know `getpoint/3` is generated from introspection but I wonder if there is some way to have a NIF function that gets the pixel data and casts the return data to the right format - thereby removing the overhead of the additional NIF call.  At least converting floats to integers for the integer based formats. Given how common `sRGB` is, this would be a beneficial change.

**Comment by akash-akya** [comment]
Hey @kipcole9, yeah that makes sense, I'll add a function.
Thanks!

**Reply by akash-akya** [reply]
@kipcole9 I am thinking of adding a function under `Vix.Vips.Image` something like `get_pixel(_image, _x, _y)`. 
Wdty?

**Comment by kipcole9** [comment]
Perfect for me - that’s exactly the name and signature of what I put in ImageSent from my iPhoneOn 20 Jul 2024, at 17:04, Akash Hiremath ***@***.***> wrote:﻿
@kipcole9 I am thinking of adding a function under Vix.Vips.Image something like get_pixel(_image, _x, _y).
Wdty?

**Comment by akash-akya** [comment]
Fixed in #157
